### ISSUE TITLE
fill-paragraph doesn't work after codeblock containing [3

### ISSUE THREAD

**Issue by spacecowboy** [jrblevin/markdown-mode#161]
Noticed this weird thing after posting a makefile containing ascii formatting codes into a codeblock.

## How to reproduce

Press `M-q` (`fill-paragraph`) on the line above and below the code block.

    aaaaaaaaaaaaaaaa aaaaaaaaaaaaaaa aaaaaaaaaaaaaaaaa aaaaaaaaaaaaaaaaaaaa
    
    ```
    [3
    ```
    
    aaaaaaaaaaaaaaaa aaaaaaaaaaaaaaa aaaaaaaaaaaaaaaaa aaaaaaaaaaaaaaaaaaaa

## Actual behavior

`fill-paragraph` does nothing on the line after the code block.

## Expected behavior

`fill-paragraph` should behave the same for both lines

## Version

```
markdown-mode is an installed package.

     Status: Installed in `~/.emacs.d/elpa/markdown-mode-20160803.1848/' (unsigned).
    Archive: n/a
    Version: 20160803.1848
   Requires: emacs-24, cl-lib-0.5
    Summary: Major mode for Markdown-formatted text
   Homepage: http://jblevins.org/projects/markdown-mode/
   Keywords: markdown github flavored markdown itex 
    Other versions: 20160803.1848 (melpa), 2.1 (melpa-stable).
```

**Comment by syohex** [comment]
Could you tell us what is problem and what is your expected behavior in more detail ?
I suppose `fill-paragraph` works after code block as below, this is not **does nothing**.

![test](https://cloud.githubusercontent.com/assets/554281/18093622/766f5058-6f0b-11e6-9c49-58a3784430b8.gif)

**Comment by jrblevin** [comment]
I was able to reproduce and (hopefully) fix this in 3f11a4a. The problem was that markdown-mode mistakenly thought the point was inside an inline link due to the unclosed square bracket. Thanks for your report and please reopen if this didn't fix the problem.

**Comment by spacecowboy** [comment]
@jrblevin that indeed seemed to do the trick. Thanks.
